app/models/program.rb

```diff
@@ -94,4 +95,29 @@ def create_difficulty_levels
                                                 color: '#EF6E69')
     true
   end
+
+  ##
+  # Check if languages string has the right format. Used as before_save.
+  #
+  def check_languages_format
+    # All white spaces are removed to allow languages to be separated by ',' and ', ' but only the first option is saved
+    return unless self.languages.present?
+    self.languages = self.languages.delete(' ')
+    # We check if every language is a valid ISO 639-1 language
+    errors.add(:languages, 'must be two letters separated by commas') && return unless
+    self.languages.match(/^$|(\A[a-z][a-z](,[a-z][a-z])*\z)/).present?
+    error = false
+    self.languages.split(',').each do |lan|
+      error = ISO_639.find(lan).nil?
+      next if !error
```

Comment:
Ah and furthermore, this won't work! It iterates anyway over the whole Enumerable, what you would need is 
``
break if error
``
(but anyway, I think my other solution would be better)